components/SingleClassView/Module/Module.tsx

```diff
@@ -0,0 +1,293 @@
+  moduleId: string;
+  title: string;
+}
+
+// eslint-disable-next-line
+const AccordionLesson = ({ lesson }: { lesson: APIModuleItem }) => {
+  const [edit, setEdit] = useState(false);
+  const api = useContext(APIContext);
+  const [title, setTitle] = useState(lesson.title);
+  const [link, setLink] = useState(lesson.link.toString());
+
+  const handleSubmit = async (): Promise<void> => {
+    const item = {
+      title: title,
+      link: link,
+      moduleId: lesson.moduleId,
+      itemId: lesson.itemId,
+    };
+    await api.updateItem(lesson.moduleId, lesson.itemId, item);
+  };
+
+  const handleCancel = async (): Promise<void> => {
+    setTitle(lesson.title);
+    setLink(lesson.link.toString());
+    setEdit(false);
+  };
+
+  // eslint-disable-next-line
+  const pencilClick = () => {
+    setEdit(!edit);
+  };
+
+  return (
+    <AccordionDetails className={styles.dropdownItem}>
+      {lesson.title}
+      <img src="/Pencil.svg" className={styles.editPencil} onClick={pencilClick} />
+      {edit ? (
+        <div className={styles.backgroundDiv}>
+          <div>Edit Lesson</div>
+          <input
+            className={`${styles.label} ${styles.classInput}`}
+            value={title}
+            onChange={(e) => setTitle(e.target.value)}
+            type="text"
+            placeholder="Lesson Title"
+          />
+          <input
+            className={`${styles.label} ${styles.classInput}`}
+            value={link}
+            onChange={(e) => setLink(e.target.value)}
+            type="text"
+            placeholder="Lesson Link"
+          />
+          <button onClick={handleCancel} className={styles.cancel}>
+            Cancel
+          </button>
+          <button onClick={handleSubmit} className={styles.submit}>
+            Save
+          </button>
+        </div>
+      ) : null}
+    </AccordionDetails>
+  );
+};
+
+type AccordionModuleProps = {
+  module: Module;
+  numModules: number;
+};
+
+const AccordionModule: React.FC<AccordionModuleProps> = ({ module, numModules }) => {
+  const api = useContext(APIContext);
+
+  const { data: lessons, error } = useSWR<APIModuleItem[]>(`${module.moduleId}`, () =>
+    api.getModuleItems(module.moduleId)
+  );
+
+  if (error) return <CustomError />;
+  if (!lessons) return <CustomLoader />;
+
+  return (
+    <Accordion>
+      <AccordionSummary className={styles.dropdownHeader} expandIcon={<ExpandMoreIcon />}>
+        {module.name}
+        <FadeMenu module={module} numModules={numModules} />
+      </AccordionSummary>
+      {lessons.map((lesson, idx) => (
+        <AccordionLesson lesson={lesson} key={`${lesson.title}-${idx}`} />
+      ))}
+    </Accordion>
+  );
+};
+
+export const ClassModule: React.FC<ModuleProps> = ({ id }) => {
+  const api = useContext(APIContext);
+  const [popup, setPopup] = useState(false);
+  const [name, setName] = useState("");
+
+  const { data: modules, error } = useSWR(`api/class/${id}/modules`, () => api.getClassModules(id));
```

Comment:
lets not use swr for this, its probably better to use `useEffect`

Reply:
Take a look at this: https://github.com/TritonSE/WA-Family-Literacy-Application/blob/dc893e77fb15e8ded0a794835f338991b2e96fe3/web/src/pages/UploadBooksPage.tsx#L34